edgelet/Cargo.lock

```diff
@@ -87,7 +87,7 @@ checksum = "cdb031dd78e28731d87d56cc8ffef4a8f36ca26c38fe2de700543e627f8a464a"
 [[package]]
 name = "aziot-cert-client-async"
 version = "0.1.0"
-source = "git+https://github.com/Azure/iot-identity-service?branch=main#dc3024ac31fecf6c36a515a6c444ba81b3f6d09f"
```

Comment:
Is it ok to target a non-release commit of iot-identity-service? If so, how will we ensure this is updated to a release commit before we release edgelet?

Reply:
@yophilav Can you elaborate on the process of ingesting IIS Commits? When do we do it for master branch and release branch?

Reply:
Generally, we only update the IIS version upon doing the release. 
1. If we know for sure that we are going to do an IIS release with the next edgelet release (not always the case), I don't see any harm in pinning the non-release IIS version. However, if we are pinning a non-release IIS version AND there is no IIS release associate w/ the edgelet, that will be an issue since edgelet will require an unofficial IIS packages. From a quick look at IIS repo, I suspect we *may* have a reason to release IIS 1.2.5 with Iotedge1.2.7 -- this decision is still up in the air for Shiproom to decide. 
2. If you want to only do an E2E test run against a particular IIS commit, please use the `aziotis.commit` variable in the E2E test pipeline. Note: if it is unspecified, it will parse and use the value from Cargo.lock.

@damonbarry  What's your thoughts if we should pin the IIS commit?

Reply:
@yophilav I think we should only pin to a release commit, to make sure we don't accidentally release against a non-release commit. But @nimanch, what will happen to this PR if it _doesn't_ include the update to our i-i-s dependency?

Reply:
@damonbarry : If we dont update the IIS Dependency, then we cannot enable the support bundle test for proxy environment since it required a change to the aziotctl check that runs as part of iotedge check. Manually Specifying aziotis.commit for scheduled E2E Run doesn't work unless we hardcode the commit in pipeline variables

On a side note, since this is a master branch (both in IIS and this repo), I am not sure what the definition is of a release-commit.  What @yophilav makes complete sense for a release branch, but not sure it does for master branch

Reply:
I think since it's master branch, it should be fine to pin it there. 
@damonbarry  Do you agree? 

For the release branch, maybe we can add a logic to have it hard coded for now? (as long as we don't forget to remove it as part of the release process).

Reply:
@yophilav @nimanch If you review the commit history in the master branch, is it clear what we've done in the past?

Reply:
Looking at the Cargo.Lock history, seems like a similar pattern was followed to increment the commit regardless of the release in master branch

![image](https://user-images.githubusercontent.com/68707510/144937634-e8309a70-ba61-4290-a077-47ebb3b890aa.png)


For release/1.2, It seems like it has been done for a mix of release creation and also for PR Updates

![image](https://user-images.githubusercontent.com/68707510/144937860-ec04d1ff-271b-4dec-a3ad-64053721fa71.png)

Reply:
@nimanch Alright I'm fine with this then